### PR TITLE
Add FairSummary

### PR DESCRIPTION
**Pull request by ChristianTackeGSI**
- [X] Main Code
- [X] Tests
- [x] Changelog
- [x] Docs

### CODE REVIEW

File: src/modules/FairSummary.cmake [hunk 1]
```diff
@@ -11,10 +11,33 @@ if(CMAKE_VERSION VERSION_LESS 3.12)
...
+---------
 
+``fair_summary_global_cxx_flags_standard()``
+============================================
+
+.. code-block:: cmake
+
+  fair_summary_global_cxx_flags_standard()
+
+If ``CMAKE_CXX_FLAGS``, or ``CMAKE_CXX_STANDARD`` are set,
```
Comment (dennisklein) [inline src/modules/FairSummary.cmake]:
```suggestion
If :variable:`CMAKE_CXX_FLAGS`, or :variable:`CMAKE_CXX_STANDARD` are set,
```

File: src/modules/FairSummary.cmake [hunk 2]
```diff
@@ -28,6 +51,21 @@ function(fair_summary_global_cxx_flags_standard)
...
+
+``fair_summary_build_types()``
+==============================
+
+.. code-block:: cmake
+
+  fair_summary_build_types()
+
+Display the available Build Types from
+``CMAKE_CONFIGURATION_TYPES`` in a nice tabular way and
```
Comment (dennisklein) [inline src/modules/FairSummary.cmake]:
```suggestion
:variable:`CMAKE_CONFIGURATION_TYPES` in a nice tabular way and
```

File: src/modules/FairSummary.cmake [hunk 3]
```diff
@@ -50,6 +88,22 @@ function(fair_summary_build_types)
...
+
+.. code-block:: cmake
+
+  fair_summary_package_dependencies()
+
+:command:`find_package2` keeps track of all the found
+packages. This function displays them (including found
+version, requested version, and install prefix) in a nice
+tabulary way.
+
```
Comment (dennisklein) [inline src/modules/FairSummary.cmake:108]:
```suggestion

The prefix column prints the value of a :variable:`<pkgname>_PREFIX`.
```

**Review by dennisklein — CHANGES_REQUESTED** [review]
See comments. Rest looks good!

File: src/modules/FairSummary.cmake [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+include(FairFormattedOutput)
+
+#[=======================================================================[.rst:
+***********
+FairSummary
+***********
+
+.. versionadded:: 0.2.0
+
+Requires CMake 3.12 or later.
```
Comment (dennisklein) [inline src/modules/FairSummary.cmake:23]:
```suggestion
Requires CMake 3.12 or later.

Depends on :module:`FairFormattedOutput`.
```

**Review by dennisklein — CHANGES_REQUESTED** [review]
one more :P

File: src/modules/FairSummary.cmake (hunk 4 above)
Comment (ChristianTackeGSI) [reply src/modules/FairSummary.cmake:23]: Rephrased a bit.

**Review by dennisklein — APPROVED** [review]

**Pull request merged**